comp110-worksheet-6/OxoBoard.cs

```diff
@@ -8,41 +8,102 @@ namespace comp110_worksheet_6
 {
 	public enum Mark { None, O, X };
 
+
 	public class OxoBoard
 	{
-		// Constructor. Perform any necessary data initialisation here.
-		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+
+        //Define 2D array for board
+        Mark[,] board;
+
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+        public OxoBoard(int width = 3, int height = 3)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            //initialize 3x3 2D char array
+            board = new Mark[3,3];
+            for(int x =0; x<width; x++)
+            {
+                for(int y=0;y<height; y++)
+                {
+                    board[x,y] = Mark.None;
+                }
+            }
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            return board[x, y];
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            //Check if x and y are within the bounds of the array.
+            if ((x>= 0 && y>=0) && (x <board.Length && y < board.Length))
```

Comment:
For a 2d array `Length` gives the width x height, so would be 9 here. You want to use `GetLength` instead.